src/ICSharpCode.SharpZipLib/Zip/ZipInputStream.cs

```diff
@@ -148,7 +148,8 @@ private static bool IsEntryCompressionMethodSupported(ZipEntry entry)
 			var entryCompressionMethod = entry.CompressionMethodForHeader;
 
 			return entryCompressionMethod == CompressionMethod.Deflated ||
-				   entryCompressionMethod == CompressionMethod.Stored;
+				   entryCompressionMethod == CompressionMethod.Stored ||
+				   entryCompressionMethod == CompressionMethod.WinZipAES;
```

Comment:
I don't think this is right - in the AES case it needs to check what the real compression method is, otherwise something like encrypted BZip2 would be included. (I think I only made it use CompressionMethodForHeader to exclude AES, if it does support AES then CompressionMethod should do?)

Reply:
Changed this to check that the entry.CompressionMethod is also Deflated or Stored.